packages/medusa/src/repositories/product.ts

```diff
@@ -155,7 +155,7 @@ export const ProductRepository = dataSource.getRepository(Product).extend({
         alias: productAlias,
         categoryAlias: "categories",
         where: joinScope,
-        joinName: "leftJoin",
+        joinName: categoryIds.length ? "innerJoin" : "leftJoin",
```

Comment:
This is incorrect. You could have categoryIds AND expand categories.

Reply:
@riqwan anything about it? Just to rephrase, I kept the same join as it was previously